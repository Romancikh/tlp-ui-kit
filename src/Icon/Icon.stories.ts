import type { Meta, StoryObj } from "@storybook/react";

import Icon from "./Icon";

const meta = {
  title: "Icon",
  component: Icon,
  parameters: {
    layout: "centered",
  },
  tags: ["autodocs"],
  argTypes: {
    color: { control: "select" },
  },
} satisfies Meta<typeof Icon>;

export default meta;
type Story = StoryObj<typeof meta>;

export const AddIcon: Story = {
  args: {
    name: "AddIcon",
  },
};

export const Colored: Story = {
  args: {
    name: "AddIcon",
    color: "success",
  },
};

export const AutoStoriesIcon: Story = {
  args: {
    name: "AutoStoriesIcon",
  },
};

export const CancelOutlinedIcon: Story = {
  args: {
    name: "CancelOutlinedIcon",
  },
};

export const CheckCircleOutlineIcon: Story = {
  args: {
    name: "CheckCircleOutlineIcon",
  },
};

export const DeleteOutlineIcon: Story = {
  args: {
    name: "DeleteOutlineIcon",
  },
};

export const EditIcon: Story = {
  args: {
    name: "EditIcon",
  },
};

export const MenuIcon: Story = {
  args: {
    name: "MenuIcon",
  },
};

export const NavigateBeforeIcon: Story = {
  args: {
    name: "NavigateBeforeIcon",
  },
};
