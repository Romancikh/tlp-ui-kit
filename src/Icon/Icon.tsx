import AddIcon from "@mui/icons-material/Add";
import AutoStoriesIcon from "@mui/icons-material/AutoStories";
import CancelOutlinedIcon from "@mui/icons-material/CancelOutlined";
import CheckCircleOutlineIcon from "@mui/icons-material/CheckCircleOutline";
import DeleteOutlineIcon from "@mui/icons-material/DeleteOutline";
import EditIcon from "@mui/icons-material/Edit";
import MenuIcon from "@mui/icons-material/Menu";
import NavigateBeforeIcon from "@mui/icons-material/NavigateBefore";

const icons = {
  AddIcon,
  AutoStoriesIcon,
  CancelOutlinedIcon,
  CheckCircleOutlineIcon,
  DeleteOutlineIcon,
  EditIcon,
  MenuIcon,
  NavigateBeforeIcon,
};

type IconProps = {
  name: keyof typeof icons;
  color?: "success" | "primary" | "error" | "secondary";
};

function Icon({ name, color }: IconProps) {
  const SelectedIcon = icons[name];
  return <SelectedIcon color={color} />;
}

export default Icon;
