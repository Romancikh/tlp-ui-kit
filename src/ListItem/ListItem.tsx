import MuiListItem from "@mui/material/ListItem";
import { PropsWithChildren } from "react";

type ListItemProps = PropsWithChildren & {
  divider?: boolean;
};

function ListItem({ children, divider }: ListItemProps) {
  return <MuiListItem divider={divider}>{children}</MuiListItem>;
}

export default ListItem;
