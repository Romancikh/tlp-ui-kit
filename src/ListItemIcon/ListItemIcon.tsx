import MuiListItemIcon from "@mui/material/ListItemIcon";
import { PropsWithChildren } from "react";

type ListItemIconProps = PropsWithChildren & {};

function ListItemIcon({ children }: ListItemIconProps) {
  return <MuiListItemIcon>{children}</MuiListItemIcon>;
}

export default ListItemIcon;
