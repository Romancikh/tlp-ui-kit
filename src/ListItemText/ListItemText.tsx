import MuiListItemText from "@mui/material/ListItemText";
import { PropsWithChildren } from "react";

type ListItemTextProps = PropsWithChildren & {
  textAlign?: "right";
};

function ListItemText({ textAlign, children }: ListItemTextProps) {
  return <MuiListItemText sx={{ textAlign }}>{children}</MuiListItemText>;
}

export default ListItemText;
