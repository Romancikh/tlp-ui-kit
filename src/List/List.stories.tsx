import type { Meta, StoryObj } from "@storybook/react";

import Icon from "../Icon/Icon";
import ListItem from "../ListItem/ListItem";
import ListItemIcon from "../ListItemIcon/ListItemIcon";
import ListItemText from "../ListItemText/ListItemText";
import List from "./List";

const meta = {
  title: "List",
  component: List,
  parameters: {
    layout: "centered",
  },
} satisfies Meta<typeof List>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  render: ({ ...args }) => (
    <List {...args}>
      <ListItem divider>
        <ListItemIcon>
          <Icon name="CheckCircleOutlineIcon" color="success" />
        </ListItemIcon>
        <ListItemText>ListItem 1</ListItemText>
      </ListItem>
      <ListItem divider>
        <ListItemIcon>
          <Icon name="CancelOutlinedIcon" color="error" />
        </ListItemIcon>
        <ListItemText>ListItem 2</ListItemText>
      </ListItem>
      <ListItem>
        <ListItemIcon>
          <Icon name="AutoStoriesIcon" color="secondary" />
        </ListItemIcon>
        <ListItemText>ListItem 3</ListItemText>
      </ListItem>
    </List>
  ),
};
