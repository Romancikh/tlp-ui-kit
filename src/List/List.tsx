import { PropsWithChildren } from "react";
import "./List.scss";

type ListProps = PropsWithChildren & {};

function List({ children }: ListProps) {
  return <ul className="list">{children}</ul>;
}

export default List;
