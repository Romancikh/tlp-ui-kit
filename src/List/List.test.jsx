import { render } from "@testing-library/react";
import Icon from "../Icon/Icon";
import ListItem from "../ListItem/ListItem";
import ListItemIcon from "../ListItemIcon/ListItemIcon";
import ListItemText from "../ListItemText/ListItemText";
import List from "./List";

test("List renders correctly", () => {
  const { container } = render(
    <List>
      <ListItem divider>
        <ListItemIcon>
          <Icon name="CheckCircleOutlineIcon" color="success" />
        </ListItemIcon>
        <ListItemText>ListItem 1</ListItemText>
      </ListItem>
      <ListItem divider>
        <ListItemIcon>
          <Icon name="CancelOutlinedIcon" color="error" />
        </ListItemIcon>
        <ListItemText>ListItem 2</ListItemText>
      </ListItem>
      <ListItem>
        <ListItemIcon>
          <Icon name="AutoStoriesIcon" color="secondary" />
        </ListItemIcon>
        <ListItemText>ListItem 3</ListItemText>
      </ListItem>
    </List>
  );
  expect(container).toMatchSnapshot();
});
