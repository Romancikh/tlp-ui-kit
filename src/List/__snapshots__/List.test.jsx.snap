// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List renders correctly 1`] = `
<div>
  <ul
    class="list"
  >
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-divider css-h8lxwn-MuiListItem-root"
    >
      <div
        class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorSuccess MuiSvgIcon-fontSizeMedium css-4b67si-MuiSvgIcon-root"
          data-testid="CheckCircleOutlineIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 7.58 10 14.17l-3.59-3.58L5 12l5 5 8-8zM12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 18c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
      </div>
      <div
        class="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          ListItem 1
        </span>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding MuiListItem-divider css-h8lxwn-MuiListItem-root"
    >
      <div
        class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorError MuiSvgIcon-fontSizeMedium css-1wxstni-MuiSvgIcon-root"
          data-testid="CancelOutlinedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm0 18c-4.41 0-8-3.59-8-8s3.59-8 8-8 8 3.59 8 8-3.59 8-8 8zm3.59-13L12 10.59 8.41 7 7 8.41 10.59 12 7 15.59 8.41 17 12 13.41 15.59 17 17 15.59 13.41 12 17 8.41z"
          />
        </svg>
      </div>
      <div
        class="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          ListItem 2
        </span>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding css-1p823my-MuiListItem-root"
    >
      <div
        class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
      >
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorSecondary MuiSvgIcon-fontSizeMedium css-kw0y8f-MuiSvgIcon-root"
          data-testid="AutoStoriesIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="m19 1-5 5v11l5-4.5V1zM1 6v14.65c0 .25.25.5.5.5.1 0 .15-.05.25-.05C3.1 20.45 5.05 20 6.5 20c1.95 0 4.05.4 5.5 1.5V6c-1.45-1.1-3.55-1.5-5.5-1.5S2.45 4.9 1 6zm22 13.5V6c-.6-.45-1.25-.75-2-1v13.5c-1.1-.35-2.3-.5-3.5-.5-1.7 0-4.15.65-5.5 1.5v2c1.35-.85 3.8-1.5 5.5-1.5 1.65 0 3.35.3 4.75 1.05.1.05.15.05.25.05.25 0 .5-.25.5-.5v-1.1z"
          />
        </svg>
      </div>
      <div
        class="MuiListItemText-root css-tlelie-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          ListItem 3
        </span>
      </div>
    </li>
  </ul>
</div>
`;
